src/main/scala/firrtl/Emitter.scala

```diff
@@ -309,11 +312,13 @@ class VerilogEmitter extends SeqTransform with Emitter {
 
      def checkArgumentLegality(e: Expression) = e match {
        case _: UIntLiteral | _: SIntLiteral | _: WRef | _: WSubField =>
+       case _ if isNot(e) =>
```

Comment:
I know you got this from me on https://github.com/freechipsproject/firrtl/pull/1264, but I realized yesterday that it's not quite right. We need to recurse into the not to check that there aren't more complex expressions:
```suggestion
       case DoPrim(Not, args, _,_) => args.foreach(checkArgumentLegality)
```
Same idea below in `checkCatArgumentLegality` which we probably ought to find a way to unify with this one